File: Grumpy Bookstore Owner.py
class Solution:
    def maxSatisfied(self, customers: List[int], grumpy: List[int], minutes: int) -> int:
        n = len(customers)
        arr = [0] * n
        total = 0
        summ = 0
        for i in range(n):
            if not grumpy[i]:
                total += customers[i]
            arr[i] = customers[i] * grumpy[i]
            if i < minutes-1:
                summ += arr[i]

        maxx = summ
        left, right = 0, minutes-1
        while right < n:
            summ += arr[right]
            maxx = max(maxx, summ)
            summ -= arr[left]
            right += 1
            left += 1
        return total + maxx

File: Grumpy Bookstore Owner.cpp
class Solution {
public:
    int maxSatisfied(vector<int>& customers, vector<int>& grumpy, int minutes) {
        int n = customers.size();
        vector<int> arr(n, 0);
        int total = 0, summ = 0;
        for (int i = 0; i < n; i++) {
            total += !grumpy[i] * customers[i];
            arr[i] = customers[i] * grumpy[i];
            summ += arr[i] * (i < minutes-1);
        }

        int maxx = summ;
        int left =0, right =minutes-1;
        while (right < n) {
            summ += arr[right];
            maxx = max(maxx, summ);
            summ -= arr[left];
            right++;
            left++;
        }
        return total + maxx;
    }
};
